source/Nuke.Common/Execution/DockerExecutor.cs

```diff
@@ -0,0 +1,166 @@
+            {
+                //File.Delete(tempFile);
+            }
+        }
+
+        private static void CopyPackageReferences()
+        {
+            Log.Information("Inlining PackageDownload references from build project");
+            var buildProject = ProjectModelTasks.ParseProject(NukeBuild.BuildProjectFile).NotNull();
+            foreach (var item in buildProject.Items)
+            {
+                if (item.ItemType == "PackageDownload")
+                {
+                    var version = item.Metadata.First(x => x.Name == "Version").EvaluatedValue.Replace("[", "").Replace("]", "");
+                    var packageId = item.EvaluatedInclude;
+                    Log.Information("Inlining {PackageId} {Version} reference from build project", packageId, version);
+                    var package = NuGetPackageResolver.GetLocalInstalledPackage(packageId, NukeBuild.BuildProjectFile, version);
+                    FileSystemTasks.CopyDirectoryRecursively(package.Directory, NukeBuild.TemporaryDirectory / "nukebuild" / packageId);
+                }
+            }
+        }
+
+        private static void PublishBuildProject(RunInDockerContainerAttribute attribute)
+        {
+            var runtime = attribute.Platform == PlatformFamily.Windows ? "win-x64" : "linux-x64";
+            Log.Information("Publishing a temporary copy of the build project (targeting {Platform}) for use within the docker container", runtime);
+            FileSystemTasks.EnsureCleanDirectory(NukeBuild.TemporaryDirectory / "nukebuild");
+            DotNetTasks.DotNetPublish(p => p
+                .SetProject(NukeBuild.BuildProjectFile)
+                .SetOutput(NukeBuild.TemporaryDirectory / "nukebuild")
+                .SetConfiguration("Release")
+                .SetVerbosity(DotNetVerbosity.Quiet)
+                .EnableNoLogo()
+                .SetRuntime(runtime)
+                .EnableSelfContained()
+            );
+        }
+
+        private static void PullImageIfRequired(RunInDockerContainerAttribute attribute)
+        {
+            if (attribute.PullImage)
+            {
+                Log.Information("Pulling image {Image}", attribute.Image);
+                DockerTasks.DockerPull(settings => settings
+                    .SetName(attribute.Image));
+            }
+        }
+
+        private static string GetEnvFile(string workingDirectory, string userProfile)
+        {
+            //todo: this will probably fail with an env var value with an ampersand in it
+            //todo: this will probably fail with an env var value with new lines in it
+            var stringBuilder = new StringBuilder();
+            var dictionaryEntries = Environment.GetEnvironmentVariables()
+                .Cast<DictionaryEntry>()
+                .Where(env =>
+                {
+                    var key = (string)env.Key;
+                    return !key.Contains(" ")
+                           && key != "USERPROFILE"
```

Comment:
This doesn't feel great, but we need to not copy these specific ones in.
I need to figure out how the teamcity docker wrapper determines which ones to inject.